src/pip/_internal/commands/list.py

```diff
@@ -285,12 +285,14 @@ def output_package_listing(
             self.output_package_listing_columns(data, header)
         elif options.list_format == "freeze":
             for dist in packages:
+                try:
+                    req_string = f"{dist.raw_name}=={dist.version}"
+                except InvalidVersion:
+                    req_string = f"{dist.raw_name}==={dist.raw_version}"
```

Comment:
The `pip list --format=freeze` does not produce identical output as `pip freeze`. For instance direct URLs are not supported. So I'm not sure emitting a === will be what user expect here, but to be honest I don't know why this output format for `pip list` exists in addition to `pip freeze`.

Reply:
Ah, are you saying that since `pip list --format=freeze` has never matched the freeze format that people may not expect anything but `name==version`? That's fair, although I have doubts on how problematic `===` would be. A) previously, pip list would crash as it had no way of handling invalid versions, and B) the verbose mode already augments the `pip list --format=freeze` format.

The alternative is that we simply hide the package with broken metadata or let pip continue crashing, and I don't think that's a particularly great outcome. We could warn that a package is hidden due to broken metadata, but given other parts of pip handle invalid versions gracefully, that would be also unexpected IMO.

Reply:
> are you saying that since `pip list --format=freeze` has never matched the freeze format that people may not expect anything but `name==version`?

Yes, that, but maybe I'm overthinking it.